Update rke2-hardening-guide based on CIS-1.24 and CIS-1.7 (#796)
rancher/rancher-docs: docs/pages-for-subheaders/rke1-hardening-guide.md

| Rancher Version | CIS Benchmark Version | Kubernetes Version           |
|-----------------|-----------------------|------------------------------|

> | Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.25 |

Reviewer: Not sure if this is intentional, but aren't these the same changes from your RKE1 PR? 
https://github.com/rancher/rancher-docs/pull/794/files

I'm ok with the changes being in one PR, but if you'd like to keep them separate I think you should undo this change and leave it to the other PR.

> | Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.24 up to v1.25 |

Author: ```suggestion
| Rancher v2.7    | Benchmark v1.7       | Kubernetes v1.25 |
```

:::note

> - Since Benchmark v1.24, check id `4.1.7 Ensure that the certificate authorities file permissions are set to 600 or more restrictive (Automated)` might fail, as /etc/kubernetes/ssl/kube-ca.pem is provisioned in 644 by default.
> - Since Benchmark v1.7 (latest), `--protect-kernel-defaults` (check id 4.2.6) parameter is not required anymore, and was replaced.

Reviewer 2: ```suggestion
- In Benchmark v1.24 and later, check id `4.1.7 Ensure that the certificate authorities file permissions are set to 600 or more restrictive (Automated)` might fail, as `/etc/kubernetes/ssl/kube-ca.pem` is set to 644 by default.
- In Benchmark v1.7, `--protect-kernel-defaults` (check id 4.2.6) parameter isn't required anymore, and was replaced.
```

Similar comment as https://github.com/rancher/rancher-docs/pull/794#discussion_r1302017920, "Should we add what the --protect-kernel-defaults parameter was replaced by? If not, should we remove mention that it was replaced?"

Author: @martyav thanks for the review. I think we should remove "and was replaced". In fact, the check id `4.2.6` has been assigned to another check, [here](https://github.com/rancher/security-scan/blob/011ad94bd1a707646e8de880cc1a780f6591b045/package/cfg/rke-cis-1.7-hardened/node.yaml#L283C4-L283C4).

  kubelet:
    extra_args:
      feature-gates: RotateKubeletServerCertificate=true

>       protect-kernel-defaults: "true"

Reviewer 3: This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.

  kubelet:
    extra_args:
      feature-gates: RotateKubeletServerCertificate=true

>       protect-kernel-defaults: true

Reviewer 3: Same as before:

> 

This needs to be inside a specific tab depending on the CIS benchmark version, because it's still needed in CIS 1.23 and CIS 1.24.
